### DISCUSSION TITLE
Why does Prisma not have a bidirectional relation?

### DISCUSSION THREAD

**Discussion by hnbnh** [prisma/prisma#10146] [Q&A] [answered]
There is an example about how to generate a one-to-one relation:

https://www.prisma.io/docs/concepts/components/prisma-schema/relations/one-to-one-relations

```prisma
model User {
  id      Int      @id @default(autoincrement())
  profile Profile?
}

model Profile {
  id     Int  @id @default(autoincrement())
  user   User @relation(fields: [userId], references: [id])
  userId Int // relation scalar field (used in the `@relation` attribute above)
}
```
In many traditional ORMs, we can declare a bidirectional relation so `User` will have a `profile` property on them, even though there is no `profile` column generated on `User` table.

My question is, why Prisma won't generate a type like so:

```typescript
export type User = {
  id: number
  profile: Profile
}
```

**Comment by austincrim** [comment, marked as answer]
Hey @hnbnh 👋, excellent question!

The generated Prisma types (the ones you would import like `import {User} from '@prisma/client'`) will only include the [*scalar*](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#model-field-scalar-types) values. However, when writing a query, Prisma will automatically include the relation types where appropriate.

```ts
let result = await prisma.user.findFirst({
  include: {
    profile: true
  }
})

// has type User & { profile: Profile }
```

If you needed to use this type elsewhere, you could define your own:
```ts
import {User, Profile} from '@prisma/client'

export type UserWithProfile = User & {
  profile: Profile
}
```

Let me know if you have further questions!

**Reply by hnbnh** [reply]
Thank you for your kind reply!

**Reply by austincrim** [reply]
Hey @hnbnh, wanted to follow up here.

My colleague informed me of a more flexible solution to this problem by using the [`Prisma` type helpers](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types#problem-using-variations-of-the-generated-model-type) like so:
```ts
import { Prisma } from '@prisma/client'

type UserWithProfile = Prisma.UserGetPayload<{
  include: { profile: true }
}>
```

This approach is a little more flexible and maintainable due to using the Prisma generated types and it allows you to use a similar syntax to a regular Prisma query.

Let me know if you have any questions about this approach!

**Reply by hnbnh** [reply]
This approach is beautiful for me! Thank you for letting me know!

**Reply by AvantaR** [reply]
I found it weird at first, but now I think it's a very convenient and flexible approach. But it would be great if we could declare in the schema file which relation should be load by default.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
